doc/howto/write_an_alphabet/dna2_derive_from_base.cpp

```diff
@@ -9,15 +9,40 @@
 #include <array>                                         // std::array
 #include <seqan3/alphabet/alphabet_base.hpp>             // alphabet_base
 #include <seqan3/alphabet/concept.hpp>                   // alphabet concept checks
+#include <seqan3/utility/char_operations/transform.hpp>  // seqan3::to_lower
 
 // derive from alphabet_base
 struct dna2 : public seqan3::alphabet_base<dna2, 2>
 {
 private:
+    // make the base class a friend so it can access the tables:
+    friend seqan3::alphabet_base<dna2, 2>;
+
     // map 0 => 'S' and 1 => 'W'
-    static constexpr char_type rank_to_char[alphabet_size] {'S', 'W'};
+    static constexpr char_type rank_to_char(rank_type const rank)
+    {
+        // via a lookup table
+        return rank_to_char_table[rank];
+        // or via an arithmetic expression
+        return rank == 1 ? 'W' : 'S';
```

Comment:
clang with `-Wunreachable-code-return` will complain about an unreachable return, but no strong feelings

Reply:
yeah I thought about this too, I wasn't sure how to express this.

Maybe two different implementations?